crates/editor/src/editor.rs

```diff
@@ -7906,7 +7906,7 @@ impl Editor {
                 .insert_blocks(
                     diagnostic_group.iter().map(|entry| {
                         let diagnostic = entry.diagnostic.clone();
-                        let message_height = diagnostic.message.lines().count() as u8;
+                        let message_height = diagnostic.message.matches('\n').count() as u8 + 1;
```

Comment:
You can see in [this Rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b98a39fec45beac5a45a9d6526d05ac9) that `"".lines().count()` returns `0`.